src/components/views/rooms/SendMessageComposer.js

```diff
@@ -536,10 +537,17 @@ export default class SendMessageComposer extends React.Component {
         }
     }
 
+    onChange = () => {
+        if (this.props.onIsEmptyChanged) {
```

Comment:
Hmm, at the moment, I think this is actually being called on _every_ change to the composer text, not only when the empty state changes. I guess this component could maintain some internal state so it only calls `onIsEmptyChanged` for an actual empty change, or alternatively you could call the props function just `onChange` so it's clear it receives the whole model for every change.